File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const flag = false;
    const n = nums.length;
    
    for(let i = 0; i < n; i++) {
        for(let j = i + 1; j < n; j++) {
            if(nums[i] == nums[j]) return true;
        }
    }
    
    return false;
};

// tc -> O(n^2);
// sc -> O(1)

File: 0217-contains-duplicate/0217-contains-duplicate.go
func containsDuplicate(nums []int) bool {
    if len(nums) <= 1 {
        return false;
    }
    
    seen := make(map[int]struct{}); // only declaration
    // seen := map[int]struct{}{}; declaring and initialization
    
    for _, v := range nums {
        if _, ok := seen[v]; ok {
            return true;
        }
        
        seen[v] = struct{}{};
    }
    
    return false;
}